fava/util/date.py

```diff
@@ -217,9 +243,76 @@ def parse_date(string):  # pylint: disable=too-many-return-statements
         quarter_first_day = datetime.date(year, (quarter - 1) * 3 + 1, 1)
         return quarter_first_day, get_next_interval(quarter_first_day,
                                                     Interval.QUARTER)
+
+    match = FY_RE.match(string)
+    if match:
+        year = int(match.group(1))
+        return get_fiscal_period(year)
+
+    match = FY_QUARTER_RE.match(string)
+    if match:
+        year, quarter = map(int, match.group(1, 2))
+        return get_fiscal_period(year, quarter)
+
     return None, None
 
 
+def month_offset(date, months):
+    """Offsets a date by a given number of months
+
+    Maintains the day, unless that day is invalid when it will
+    raise a ValueError
+
+    """
+
+    year_delta, month = divmod(date.month - 1 + months, 12)
+
+    return date.replace(year=date.year + year_delta, month=month + 1)
+
+
+def get_fiscal_period(year, quarter=None, fye=None):
+    """Calculates fiscal periods
+
+    Uses the fava option "fiscal-year-end" which should be in "%m-%d" format.
+    Defaults to calendar year [12-31]
+
+    Args:
+        year: An interger year
+        quarter: one of [None, 1, 2, 3 or 4]
+        fye: End date for period in "%m-%d" format
+
+    Returns:
+        A tuple (start, end) of dates.
+
+    """
+    if fye is None:
+        from flask import g
+        fye = g.ledger.fava_options['fiscal-year-end']
```

Comment:
I'd prefer it if `util.date` didn't depend on the active request context - better pass the fiscal year end as an explicit argument to the functions in this module

Reply:
I agree, but from groking the code you'd need to inject the arguments in the `FavaLedger.filter()` method and adjust the signatures on all the `Filter` classes - I didn't want to make that call!

Am happy to refactor if you can give me some guidance on what context you want passed down to the filters - am guessing the `fava_options` object for future expansion?

Reply:
I tried moving the context access to the `TimeFilter` class but then the filter tests started failing as there is no valid context.  
Have just created another pull request at #807 which makes the Filters stateful and would allow for generic configurable filters - will wait to hear your input on that before continuing here...